arq/worker.py

```diff
@@ -450,9 +460,15 @@ def run(self) -> None:
                 if e.defer_score:
                     incr_score = e.defer_score + (timestamp_ms() - score)
                 self.jobs_retried += 1
-            elif self.retry_jobs and isinstance(e, asyncio.CancelledError):
-                logger.info('%6.2fs ↻ %s cancelled, will be run again', t, ref)
-                self.jobs_retried += 1
+            elif isinstance(e, asyncio.CancelledError):
+                if job_id in self.abort_tasks:
+                    logger.info('%6.2fs 🛇  %s aborted', t, ref)
+                    result = e
+                    finish = True
+                    self.abort_tasks.remove(job_id)
+                elif self.retry_jobs:
```

Comment:
technically the `elif, else` logic has changed here.

I think it should be:
```suggestion
            elif isinstance(e, asyncio.CancelledError) and (self.retry_jobs or job_id in self.abort_tasks):
                if job_id in self.abort_tasks:
                    logger.info('%6.2fs 🛇  %s aborted', t, ref)
                    result = e
                    finish = True
                    self.abort_tasks.remove(job_id)
                else:
```

Reply:
I fix it in this way:

```python
elif isinstance(e, asyncio.CancelledError) and job_id in self._aborting_tasks:
    logger.info('%6.2fs 🛇  %s aborted', t, ref)
    result = e
    finish = True
    self._aborting_tasks.remove(job_id)
   # redis key will be removed at next scan
elif isinstance(e, asyncio.CancelledError) and self.retry_jobs:
    logger.info('%6.2fs ↻ %s cancelled, will be run again', t, ref)
    self.jobs_retried += 1
```

Reply:
I think this is less clear and duplicates the isinstance check.